datafusion/tests/test_context.py

```diff
@@ -282,10 +282,11 @@ def test_dataset_filter_nested_data(ctx):
 
     # This filter will not be pushed down to DatasetExec since it
     # isn't supported
-    df = df.select(
+    df = df.filter(column("nested_data")["b"] > literal(5))\
```

Comment:
Move the filter before the select due to https://github.com/apache/arrow-datafusion/issues/5949